core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -17,16 +17,28 @@
 
 package kafka.utils
 
-import org.apache.log4j.Logger
+import org.slf4j.{LoggerFactory, MarkerFactory}
+
+object Log4jControllerInstantiator {
+  try {
+    val log4jController = Class.forName("kafka.utils.Log4jController").asInstanceOf[Class[Object]]
+    val instance = log4jController.newInstance()
+    CoreUtils.registerMBean(instance, "kafka:type=kafka.Log4jController")
+  } catch {
+    case e: ClassNotFoundException => {}
+  }
+}
 
 trait Logging {
   val loggerName = this.getClass.getName
-  lazy val logger = Logger.getLogger(loggerName)
+  lazy val logger = LoggerFactory.getLogger(loggerName)
 
   protected var logIdent: String = null
 
+  val fatalMarker = MarkerFactory.getMarker("FATAL")
+
```

Comment:
Due to the fact that with a method call I'd have to use booleans and possibly synchronization or other clever tricks to avoid concurrency issues, I'd stay with a simpler alternative, like renaming the object itself:

`object EnsureControllerRegistered {
  try {
    val log4jController = Class.forName("kafka.utils.Log4jController").asInstanceOf[Class[Object]]
    val instance = log4jController.newInstance()
    CoreUtils.registerMBean(instance, "kafka:type=kafka.Log4jController")
  } catch {
    case e: ClassNotFoundException => {}
  }
}

/* ... */

private val singletonRegistrar = EnsureControllerRegistered`